charms/knative-operator/tests/unit/test_charm.py

```diff
@@ -135,53 +144,100 @@ def test_apply_resources_exception(
-                    "POD_NAME": harness.model.app.name,
-                    "SYSTEM_NAMESPACE": harness.model.name,
-                    "METRICS_DOMAIN": "knative.dev/operator",
-                    "CONFIG_LOGGING_NAME": "config-loggig",
-                    "CONFIG_OBSERVABILITY_NAME": "config-observability",
-                },
-            }
-        },
+    harness_events = {
+        KNATIVE_OPERATOR: harness.charm.on.knative_operator_pebble_ready,
+        KNATIVE_OPERATOR_WEBHOOK: harness.charm.on.knative_operator_webhook_pebble_ready,
     }
-    updated_plan = harness.get_container_pebble_plan("knative-operator").to_dict()
+    harness_events[container_name].emit(container_name)
+    assert harness.get_container_pebble_plan(container_name)._services is not None
+
+    # TODO: This could be extracted and made cleaner
+    expected_plan = {}
+    if container_name == "knative-operator":
+        expected_plan = {
+            "services": {
+                KNATIVE_OPERATOR: {
+                    "summary": "entrypoint of the knative-operator image",
+                    "startup": "enabled",
+                    "override": "replace",
+                    "command": "/ko-app/operator",
+                    "environment": {
+                        "POD_NAME": harness.model.app.name,
+                        "SYSTEM_NAMESPACE": harness.model.name,
+                        "METRICS_DOMAIN": "knative.dev/operator",
+                        "CONFIG_LOGGING_NAME": "config-logging",
+                        "CONFIG_OBSERVABILITY_NAME": "config-observability",
+                    },
+                }
+            },
+        }
+    elif container_name == "knative-operator-webhook":
+        expected_plan = {
+            "services": {
+                container_name: {
+                    "summary": "entrypoint of the knative-operator-webhook image",
+                    "startup": "enabled",
+                    "override": "replace",
+                    "command": "/ko-app/webhook",
+                    "environment": {
+                        "POD_NAME": harness.model.app.name,
+                        "SYSTEM_NAMESPACE": harness.model.name,
+                        "METRICS_DOMAIN": "knative.dev/operator",
+                        "CONFIG_LOGGING_NAME": "config-logging",
+                        "CONFIG_OBSERVABILITY_NAME": "config-observability",
+                        "WEBHOOK_NAME": "operator-webhook",
+                        "WEBHOOK_PORT": "8443",
+                    },
+                }
+            },
+        }
+    else:
+        raise ValueError(f"Unknown container name: {container_name}")
+    updated_plan = harness.get_container_pebble_plan(container_name).to_dict()
     assert expected_plan == updated_plan
 
-    service = harness.model.unit.get_container("knative-operator").get_service("/ko-app/operator")
+    service = harness.model.unit.get_container(container_name).get_service(container_name)
```

Comment:
Did you leave the same name for the service and the container?

Reply:
yes, I used the container's name as the service name, but agreed this looks odd.  I'll refactor to make it clear